crates/activate/src/lib.rs

```diff
@@ -587,6 +568,8 @@ fn task_changes<'a>(
             upsert: Some(recovery_spec),
             delete: String::new(),
         }));
+
+        active = true;
```

Comment:
nit: would it make sense to move this up to line 495? IIUC, it's always true unless we're deleting the shard. Maybe rename to `is_upsert`?

Reply:
removed in favor of using an iter().any(matches!(..))